shared/libraries/websocket_streaming/src/streaming_client.cpp

```diff
@@ -38,11 +39,46 @@ StreamingClient::StreamingClient(const ContextPtr& context, const std::string& h
     , signalContainer(logCallback)
     , useRawTcpConnection(useRawTcpConnection)
 {
+    startBackgroundContext();
+}
+
+void daq::websocket_streaming::StreamingClient::startBackgroundContext()
+{
+    clientBackgroundThread = std::thread(
+        [this]()
+        {
+            using namespace boost::asio;
+            executor_work_guard<io_context::executor_type> workGuard(backgroundContext.get_executor());
+            backgroundContext.run();
+        }
+    );
+}
+
+void StreamingClient::stopBackgroundContext()
+{
+    backgroundContext.stop();
+    if (clientBackgroundThread.get_id() != std::this_thread::get_id())
+    {
+        if (clientBackgroundThread.joinable())
+        {
+            clientBackgroundThread.join();
+            LOG_I("Websocket streaming client background thread joined");
+        }
+        else
+        {
+            LOG_W("Websocket streaming client background thread is not joinable");
+        }
+    }
+    else
+    {
+        LOG_C("Websocket streaming client background thread cannot join itself");
```

Comment:
An additional exception could be thrown here, but it may not be necessary since the program will call std::terminate in thread destructor if the thread is not joined, as in this case. The critical level log message here just helps identify which thread caused the termination.